src/main/java/uk/gov/hmcts/reform/slc/services/steps/maptoletter/LetterMapper.java

```diff
@@ -15,18 +19,28 @@
 public class LetterMapper {
 
     private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static final Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
 
     public Letter from(IMessage msg) {
         try {
-            Letter letter = objectMapper.readValue(msg.getBody(), Letter.class);
-            if (letter != null) {
+            return validate(objectMapper.readValue(msg.getBody(), Letter.class));
+        } catch (IOException exc) {
+            throw new InvalidMessageException("Unable to deserialize message " + msg.getMessageId(), exc);
+        }
+    }
+
+    private Letter validate(Letter letter) throws InvalidMessageException {
+        if (letter != null) {
+            Set<ConstraintViolation<Letter>> violations = validator.validate(letter);
```

Comment:
I am not sure why you need to validate the letter.Are you expecting message to be modified once it is persisted into queue?

Reply:
It can be once it is resubmitted from the dead letter queue

Reply:
Also code could be rewritten as below instead of if else if else
```
private Letter validate(Letter letter) throws InvalidMessageException {
		if(letter==null){
			throw new InvalidMessageException("Invalid message body");
		}

		Set<ConstraintViolation<Letter>> violations = validator.validate(letter);

		if(!violations.isEmpty()){
			throw new InvalidMessageException("Invalid message body");
         }

        return letter;
 }

```

Reply:
looks like we need pmd rule enabled